openedx/stanford/lms/djangoapps/courseware/views.py

```diff
@@ -0,0 +1,265 @@
+from datetime import datetime
+import json
+
+from django.conf import settings
+from django.http import Http404
+from django.http import HttpResponseNotFound, HttpResponseServerError
+from django.utils.translation import ugettext as _
+
+from pytz import timezone
+
+from analyticsclient.client import Client
+from analyticsclient.exceptions import NotFoundError, InvalidRequestError, TimeoutError
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
+
+from courseware.access import has_access
+from courseware.courses import get_course_with_access
+from courseware.models import CoursePreference
+from shoppingcart.utils import is_shopping_cart_enabled
+from student.models import UserProfile
+from util.date_utils import get_time_display
+from util.json_request import JsonResponse
+
+
+def get_context_for_index_bulk_op():
+    context = {
+        'analytics_url': settings.ANALYTICS_DATA_URL,
+
+    }
+    return context
+
+
+def get_context_for_course_about(registered, user, course, course_key):
+    regularly_registered = (
+        registered
+        and
+        UserProfile.has_registered(user)
+    )
+    # only allow course sneak peek if
+    # 1) within enrollment period
+    # 2) course specifies it's okay
+    # 3) user is not a registered user.
+    sneakpeek_allowed = (
+        has_access(user, 'within_enrollment_period', course)
+        and
+        CoursePreference.course_allows_nonregistered_access(course_key)
+        and
+        not UserProfile.has_registered(user)
+    )
+    context = {
+        'is_shopping_cart_enabled': is_shopping_cart_enabled,
+        'regularly_registered': regularly_registered,
+        'sneakpeek_allowed': sneakpeek_allowed,
+    }
+    return context
+
+
+def get_analytics_answer_dist(request):
```

Comment:
These big inline_analytics methods could arguably be moved out into their own Django app.
I'm not worried about that for now.
For the time being, we're just mimicking the same directory structure as was previously used.